### PR TITLE
Remove link to self-attention modules.

### PR DESCRIPTION
**Pull request by himkt**
Follow-up for #3941.

I was looking for self-attention modules to find that these were moved to allennlp-model.
I removed the links with no destination.

### CODE REVIEW

**Comment by epwalsh** [comment]
Thanks @himkt! It looks like there are some other things in that docstring list that are not up-to-date. Would you mind fixing those as well as part of this PR?

For example, paths like `allennlp.modules.pass_through_encoder.PassThroughEncoder` should really be `allennlp.modules.seq2seq_encoder.PassThroughEncoder`. And there are also a few encoders missing from that list, such as `ComposeEncoder`.

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 1]
```diff
@@ -6,17 +6,17 @@
 
 The available Seq2Seq encoders are
 
-- `"gru"` : https://pytorch.org/docs/master/nn.html#torch.nn.GRU
-- `"lstm"` : https://pytorch.org/docs/master/nn.html#torch.nn.LSTM
-- `"rnn"` : https://pytorch.org/docs/master/nn.html#torch.nn.RNN
+- `"gru"` : https://pytorch.org/docs/master/generated/torch.nn.GRU.html#torch.nn.GRU
```
Comment (himkt) [inline allennlp/modules/seq2seq_encoders/__init__.py:9]: fde5404bddb
Fix link to PyTorch modules.
BTW, I'm wondering if these should refer to `allennlp.modules.seq2seq_encoders.pytorch_seq2seq_wrapper.*`.

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 2]
```diff
@@ -6,17 +6,17 @@
...
+- `"lstm"` : https://pytorch.org/docs/master/generated/torch.nn.LSTM.html#torch.nn.LSTM
+- `"rnn"` : https://pytorch.org/docs/master/generated/torch.nn.RNN.html#torch.nn.RNN
 - `"augmented_lstm"` : allennlp.modules.augmented_lstm.AugmentedLstm
 - `"alternating_lstm"` : allennlp.modules.stacked_alternating_lstm.StackedAlternatingLstm
 - `"alternating_highway_lstm"` : allennlp.modules.stacked_alternating_lstm.StackedAlternatingLstm (GPU only)
-- `"stacked_self_attention"` : allennlp.modules.stacked_self_attention.StackedSelfAttentionEncoder
-- `"multi_head_self_attention"` : allennlp.modules.multi_head_self_attention.MultiHeadSelfAttention
-- `"pass_through"` : allennlp.modules.pass_through_encoder.PassThroughEncoder
-- `"feedforward"` : allennlp.modules.feedforward_encoder.FeedforwardEncoder
+- `"pass_through"` : allennlp.modules.seq2seq_encoders.PassThroughEncoder
```
Comment (himkt) [inline allennlp/modules/seq2seq_encoders/__init__.py:15]: 0010d00e079b74
Add `ComposeEncoder` and `GatedCnnEncoder`.

Comment (himkt) [inline allennlp/modules/seq2seq_encoders/__init__.py:15]: 490276d5842
Update path: `allennlp.modules.{pass_through,feedforward}_encoder` -> `allennlp.modules.seq2seq_encoders`

**Comment by himkt** [comment]
@epwalsh
I revised the PR, PTAL.

File: CHANGELOG.md [hunk 3]
```diff
@@ -13,6 +13,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Use slower tqdm intervals when output is being piped or redirected.
 - Fixed testing models that only return a loss when they are in training mode.
 - Fixed a bug in `FromParams` that causes silent failure in case of the parameter type being Optional[Union[...]].
+- Cleanup/fix to encoder modules in `allennlp/models/seq2seq_encoders/__init__.py`.
```
Comment (epwalsh) [inline CHANGELOG.md:16]:
```suggestion
- Cleanup/fix to module docstring in `allennlp/models/seq2seq_encoders/__init__.py`.
```

File: allennlp/modules/seq2seq_encoders/__init__.py (hunk 1 above)
Comment (epwalsh) [reply allennlp/modules/seq2seq_encoders/__init__.py:9]: Yea, I think agree with you there. Feel free to make that change if you want.

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 4]
```diff
@@ -6,17 +6,17 @@
...
 The available Seq2Seq encoders are
 
-- `"gru"` : https://pytorch.org/docs/master/nn.html#torch.nn.GRU
-- `"lstm"` : https://pytorch.org/docs/master/nn.html#torch.nn.LSTM
-- `"rnn"` : https://pytorch.org/docs/master/nn.html#torch.nn.RNN
+- `"gru"` : https://pytorch.org/docs/master/generated/torch.nn.GRU.html#torch.nn.GRU
+- `"lstm"` : https://pytorch.org/docs/master/generated/torch.nn.LSTM.html#torch.nn.LSTM
+- `"rnn"` : https://pytorch.org/docs/master/generated/torch.nn.RNN.html#torch.nn.RNN
 - `"augmented_lstm"` : allennlp.modules.augmented_lstm.AugmentedLstm
 - `"alternating_lstm"` : allennlp.modules.stacked_alternating_lstm.StackedAlternatingLstm
```
Comment (epwalsh) [inline allennlp/modules/seq2seq_encoders/__init__.py:13]: I think this should be:

```suggestion
- `"alternating_lstm"` : allennlp.modules.seq2seq_encoders.StackedAlternatingLstmSeq2SeqEncoder
```

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 5]
```diff
@@ -6,17 +6,17 @@
 
 The available Seq2Seq encoders are
 
-- `"gru"` : https://pytorch.org/docs/master/nn.html#torch.nn.GRU
-- `"lstm"` : https://pytorch.org/docs/master/nn.html#torch.nn.LSTM
-- `"rnn"` : https://pytorch.org/docs/master/nn.html#torch.nn.RNN
+- `"gru"` : https://pytorch.org/docs/master/generated/torch.nn.GRU.html#torch.nn.GRU
+- `"lstm"` : https://pytorch.org/docs/master/generated/torch.nn.LSTM.html#torch.nn.LSTM
+- `"rnn"` : https://pytorch.org/docs/master/generated/torch.nn.RNN.html#torch.nn.RNN
 - `"augmented_lstm"` : allennlp.modules.augmented_lstm.AugmentedLstm
```
Comment (epwalsh) [inline allennlp/modules/seq2seq_encoders/__init__.py:12]: I think this should be:

```suggestion
- `"augmented_lstm"` : allennlp.modules.seq2seq_encoders.AugmentedLstmSeq2SeqEncoder
```

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 6]
```diff
@@ -6,17 +6,17 @@
...
-- `"stacked_self_attention"` : allennlp.modules.stacked_self_attention.StackedSelfAttentionEncoder
-- `"multi_head_self_attention"` : allennlp.modules.multi_head_self_attention.MultiHeadSelfAttention
-- `"pass_through"` : allennlp.modules.pass_through_encoder.PassThroughEncoder
-- `"feedforward"` : allennlp.modules.feedforward_encoder.FeedforwardEncoder
+- `"pass_through"` : allennlp.modules.seq2seq_encoders.PassThroughEncoder
+- `"feedforward"` : allennlp.modules.seq2seq_encoders.FeedForwardEncoder
 - `"pytorch_transformer"` : allennlp.modules.seq2seq_encoders.PytorchTransformer
+- `"compose"` : allennlp.modules.seq2seq_encoders.ComposeEncoder
+- `"gated-cnn-encoder"` : allennlp.momdules.seq2seq_encoders.GatedCnnEncoder
 """
```
Comment (epwalsh) [inline allennlp/modules/seq2seq_encoders/__init__.py:20]: I think we're also missing "stacked_bidirectional_lstm"

File: allennlp/modules/seq2seq_encoders/__init__.py [hunk 7]
```diff
@@ -6,17 +6,17 @@
...
 
-- `"gru"` : https://pytorch.org/docs/master/nn.html#torch.nn.GRU
-- `"lstm"` : https://pytorch.org/docs/master/nn.html#torch.nn.LSTM
-- `"rnn"` : https://pytorch.org/docs/master/nn.html#torch.nn.RNN
+- `"gru"` : https://pytorch.org/docs/master/generated/torch.nn.GRU.html#torch.nn.GRU
+- `"lstm"` : https://pytorch.org/docs/master/generated/torch.nn.LSTM.html#torch.nn.LSTM
+- `"rnn"` : https://pytorch.org/docs/master/generated/torch.nn.RNN.html#torch.nn.RNN
 - `"augmented_lstm"` : allennlp.modules.augmented_lstm.AugmentedLstm
 - `"alternating_lstm"` : allennlp.modules.stacked_alternating_lstm.StackedAlternatingLstm
 - `"alternating_highway_lstm"` : allennlp.modules.stacked_alternating_lstm.StackedAlternatingLstm (GPU only)
```
Comment (epwalsh) [inline allennlp/modules/seq2seq_encoders/__init__.py:14]: I don't think this one exists

Comment (himkt) [reply allennlp/modules/seq2seq_encoders/__init__.py:14]: Thank you for correcting me. As you said, [`StackedAlternatingLstm`](http://docs.allennlp.org/master/api/modules/stacked_alternating_lstm/#stackedalternatinglstm) is not a Seq2seqEncoder. I remove it.

File: allennlp/modules/seq2seq_encoders/__init__.py (hunk 1 above)
Comment (himkt) [reply allennlp/modules/seq2seq_encoders/__init__.py]: Updated in c0e440a.

File: allennlp/modules/seq2seq_encoders/__init__.py (hunk 6 above)
Comment (himkt) [reply allennlp/modules/seq2seq_encoders/__init__.py:20]: Thank you for catching, added in 1d6550c.

**Pull request merged**